src/main/java/emu/grasscutter/game/entity/GameEntity.java

```diff
@@ -160,8 +164,8 @@ public void damage(float amount, int killerId) {
         // Invoke entity damage event.
         EntityDamageEvent event = new EntityDamageEvent(this, amount, this.getScene().getEntityById(killerId));
         event.call();
-        if (event.isCanceled()) {
-            return; // If the event is canceled, do not damage the entity.
+        if (event.isCanceled() || !hasFightProperty(FightProperty.FIGHT_PROP_CUR_HP)) {
```

Comment:
Check should be bundled in with the previous `this.getFightProperties() == null` rather than making an `EntityDamageEvent`.

Reply:
fixed